File: ApiCallTest/app/src/main/java/com/example/raluc/apicalltest/ui/MainActivity.java
package com.example.raluc.apicalltest.ui;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.os.Parcelable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.example.raluc.apicalltest.R;
import com.example.raluc.apicalltest.viewModels.MainViewModel;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MAinActivity";
    private MainViewModel mainViewModel;
    private Button button;
    private EditText et_name;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d(TAG, "onCreate");


        mainViewModel = ViewModelProviders.of(this).get(MainViewModel.class);
        et_name = findViewById(R.id.et_artist);


    }

    public void searchArtist(View view) {

        Intent intent = new Intent(this, SearchResults.class);
        intent.putExtra("artist", et_name.getText().toString());
        startActivity(intent);
    }
}

File: ContentProviders/app/src/main/java/com/example/raluc/contentproviders/ui/MainActivity.kt
package com.example.raluc.contentproviders.ui

import android.content.ContentValues
import android.database.Cursor
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.app.LoaderManager
import android.support.v4.content.CursorLoader
import android.support.v4.content.Loader
import android.view.View
import android.widget.Toast
import com.example.raluc.contentproviders.R
import com.example.raluc.contentproviders.common.URL_LOADER
import com.example.raluc.contentproviders.providers.MyContactProviderContract
import kotlinx.android.synthetic.main.activity_main.*
import java.lang.NullPointerException

class MainActivity : AppCompatActivity() , LoaderManager.LoaderCallbacks<Cursor> {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        supportLoaderManager.initLoader(URL_LOADER, null, this)
    }

    override fun onCreateLoader(loaderId: Int, bundle: Bundle?): Loader<Cursor> {
        return when(loaderId){
            URL_LOADER -> CursorLoader(
                this, // Context
            MyContactProviderContract.MyContentProviderEntry.CONTENT_URI, // table to query
            null, // projection, includes all the columns needed
            null, // selection clause
            null, // selection args
            null // sort order
            )

            else -> throw NullPointerException ()  // don't do in practice. Handle the exception
        }

    }

    override fun onLoadFinished(loader: Loader<Cursor>, cursor: Cursor?) {
        Toast.makeText(this, "Load done", Toast.LENGTH_LONG).show()
    }

    override fun onLoaderReset(p0: Loader<Cursor>) {

    }

    fun insertContact(view: View){
        val contentValues = ContentValues()
        contentValues.put(MyContactProviderContract.MyContentProviderEntry.COLUMN_CONTACT_NAME
        , etContactName.text.toString())
        contentValues.put(MyContactProviderContract.MyContentProviderEntry.COLUMN_CONTACT_RELATIONSHIP
            , etContactRelationship.text.toString())
        contentValues.put(MyContactProviderContract.MyContentProviderEntry.COLUMN_CONTACT_PRIMARY_NUMBER
            , etContactPrimaryNumber.text.toString())

        contentValues.put(MyContactProviderContract.MyContentProviderEntry.COLUMN_CONTACT_SECONDARY_NUMBER
            , etContactSecondaryNumber.text.toString()
        )
        contentResolver.insert(MyContactProviderContract.MyContentProviderEntry.CONTENT_URI, contentValues)
    }
}
